packages/wp-checkout/src/components/checkout.js

```diff
@@ -0,0 +1,264 @@
+import { usePaymentMethodData, usePaymentMethod, usePaymentMethodId } from '../lib/payment-methods';
+import theme from '../theme';
+import CheckoutNextStepButton from './checkout-next-step-button';
+import CheckoutReviewOrder from './checkout-review-order';
+import CheckoutSubmitButton from './checkout-submit-button';
+
+export default function Checkout( {
+	locale,
+	items,
+	total,
+	onChangeBillingContact,
+	availablePaymentMethods,
+	onSuccess,
+	onFailure,
+	successRedirectUrl,
+	failureRedirectUrl,
+	upSell,
+	checkoutHeader,
+	className,
+} ) {
+	const localize = localizeFactory( locale );
+	const [ stepNumber, setStepNumber ] = useState( 1 );
+
+	return (
+		<ThemeProvider theme={ theme }>
+			<CheckoutProvider
+				localize={ localize }
+				items={ items }
+				total={ total }
+				onSuccess={ onSuccess }
+				onFailure={ onFailure }
+				successRedirectUrl={ successRedirectUrl }
+				failureRedirectUrl={ failureRedirectUrl }
+			>
+				<Container className={ joinClasses( [ className, 'checkout' ] ) }>
+					<LeftColumn>
+						<div>
+							{ checkoutHeader || <PageTitle>{ localize( 'Complete your purchase' ) }</PageTitle> }
+						</div>
+						<PaymentMethodsStep
+							availablePaymentMethods={ availablePaymentMethods }
+							setStepNumber={ setStepNumber }
+							isActive={ stepNumber === 1 }
+							isComplete={ stepNumber > 1 }
+						/>
+						<BillingDetailsStep
+							setStepNumber={ setStepNumber }
+							isActive={ stepNumber === 2 }
+							isComplete={ stepNumber > 2 }
+							onChangeBillingContact={ onChangeBillingContact }
+						/>
+						<ReviewOrderStep
+							setStepNumber={ setStepNumber }
+							isActive={ stepNumber === 3 }
+							isComplete={ stepNumber > 3 }
+						/>
+						<CheckoutSubmitButton isActive={ stepNumber === 3 } />
+						{ upSell && <div>{ upSell }</div> }
+					</LeftColumn>
+				</Container>
+			</CheckoutProvider>
+		</ThemeProvider>
+	);
+}
+
+Checkout.propTypes = {
+	className: PropTypes.string,
+	locale: PropTypes.string.isRequired,
+	items: PropTypes.array.isRequired,
+	total: PropTypes.object.isRequired,
+	onChangeBillingContact: PropTypes.func,
+	availablePaymentMethods: PropTypes.arrayOf( PropTypes.string ),
+	onSuccess: PropTypes.func.isRequired,
+	onFailure: PropTypes.func.isRequired,
+	successRedirectUrl: PropTypes.string.isRequired,
+	failureRedirectUrl: PropTypes.string.isRequired,
+	reviewContent: PropTypes.element,
+	reviewContentCollapsed: PropTypes.element,
+	upSell: PropTypes.element,
+	checkoutHeader: PropTypes.element,
+	orderReviewTOS: PropTypes.element,
+	orderReviewFeatures: PropTypes.element,
+};
+
+function PaymentMethodsStep( { setStepNumber, isActive, isComplete, availablePaymentMethods } ) {
+	const localize = useLocalize();
+	const paymentMethod = usePaymentMethod();
+	const [ , setPaymentMethod ] = usePaymentMethodId();
+
+	// We must always display both the active and inactive version to keep their
```

Comment:
Let's take a look at this after it merges. I don't like that we're duplicating the Step header along with the content. Ideally, we're just toggling the view of the summary or main content area inside the step.

Reply:
Agreed. This was the result of a poor design decision on my part: I had originally forgotten that we need to have both displayed at the same time.

Reply:
Fixed by #36896